scilpy/io/streamlines.py

```diff
@@ -115,6 +118,145 @@ def load_tractogram_with_reference(parser, args, filepath, arg_name=None):
+    files_to_verify: List[str]
+        List of files that should be compatible with the reference sft. Files
+        can be either other tractograms or nifti files (ex: masks).
+    parser: argument parser
+        Will raise an error if a file is not compatible.
+    args: Namespace
+        Should contain a args.reference if any file is a .tck, and possibly a
+        args.bbox_check (set to True by default).
+    """
+    save_ref = args.reference
+
+    for file in files_to_verify:
+        if file is not None:
+            _, ext = os.path.splitext(file)
+            if ext in ['.trk', '.tck', '.fib', '.vtk', '.dpy']:
+                # Cheating ref because it may send a lot of warning if loading
+                # many trk with ref (reference was maybe added only for some
+                # of these files)
+                if ext == '.trk':
+                    args.reference = None
+                else:
+                    args.reference = save_ref
+                mask = load_tractogram_with_reference(parser, args, file)
+            else:  # should be a nifti file.
+                mask = file
+            compatible = is_header_compatible(ref_sft, mask)
+            if not compatible:
+                parser.error("Reference tractogram incompatible with {}"
+                             .format(file))
+
+
+def load_dps_files_as_dps(parser, dps_files, sft, keys=None, overwrite=False):
+    """
+    Load dps information. They must be scalar values.
+
+    Parameters
+    ----------
+    parser: parser
+    dps_files: list[str]
+        Either .npy or .txt files.
+    sft: StatefulTractogram
+    keys: list[str]
+        If None, use the filenames as keys.
+    overwrite: bool
+        If True, allow overwriting an existing dps key.
+
+    Returns
+    -------
+    sft: StatefulTractogram
+        The modified SFT. (Note that it is modified in-place even if the
+        returned variable is not used!)
+    new_keys: list[str]
+        Added keys.
+    """
+    if keys is not None and len(keys) != len(dps_files):
+        parser.error("You must provide one key name per dps file.")
+
+    new_keys = []
+    for i, file in enumerate(dps_files):
+        if keys is None:
+            name = os.path.basename(file)
+            key, ext = os.path.splitext(name)
+        else:
+            key = keys[i]
+
+        if key in sft.data_per_streamline and not overwrite:
+            parser.error("Key {} already exists in your tractogram's dps. "
+                         "You must allow overwriting keys."
+                         .format(key))
+
+        data = np.squeeze(load_matrix_in_any_format(file))
+        if len(data) != len(sft):
+            parser.error('Wrong dps size in file {}. Expected one value per '
+                         'streamline ({}) but got {} values!'
+                         .format(file, len(sft), len(data)))
+
+        new_keys.append(key)
+        sft.data_per_streamline[key] = data
+    return sft, new_keys
```

Comment:
why do you need new_keys ?

Reply:
In the main, I want to work on the newly added dps.

In the case where the key name is taken from the filename, it is not trivial to know which keys you just added (the names come from `splitext(filenames)`).

Maybe the SFT already contained others keys, so we can't just take `sft.data_per_streamlines.keys())` either.

Reply:
:+1: